src-bin/toy_github_delegate.ml

```diff
@@ -26,32 +46,15 @@ let repo_docdir_path_from_doc_uri uri =
 let publish_doc_gh_pages uri name version docdir =
   Fpath.of_string docdir
   >>= fun docdir -> repo_docdir_path_from_doc_uri uri
-  >>= fun dir -> (Topkg_care.Delegate.publish_in_git_branch
+  >>= fun dir -> repo_and_owner_of_uri uri
```

Comment:
Maybe you could simply derive the repo and owner from the `doc` uri:  `$owner.github/$repo/` since that's the only thing the delegate will provide you with (see `topkg help delegate`)